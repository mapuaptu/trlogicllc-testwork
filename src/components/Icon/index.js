import {
  mdiPlusOutline,
  mdiDeleteOutline,
  mdiPencilOutline,
  mdiGithub,
} from '@mdi/js';

const icons = {
  'plus-outline': mdiPlusOutline,
  'delete-outline': mdiDeleteOutline,
  'pencil-outline': mdiPencilOutline,
  github: mdiGithub,
};

export default {
  name: 'Icon',
  props: {
    icon: {
      type: String,
      default: '',
    },
    fill: {
      type: String,
      default: '#fff',
    },
    width: {
      type: Number,
      default: 24,
    },
    height: {
      type: Number,
      default: 24,
    },
  },
  computed: {
    iconPath() {
      return icons[this.icon];
    },
  },
};
