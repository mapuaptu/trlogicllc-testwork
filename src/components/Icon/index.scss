@import '@/styles/_variables.scss';

.icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
