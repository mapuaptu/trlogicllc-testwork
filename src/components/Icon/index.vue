<template>
  <div :class="$style.icon">
    <svg
      :width="width"
      :height="height"
      :viewbox="`0 0 ${width} ${height}`"
    >
      <path
        :d="iconPath"
        :fill="fill"
      />
    </svg>
  </div>
</template>

<script src="./index.js"></script>
<style src="./index.scss" lang="scss" module></style>
