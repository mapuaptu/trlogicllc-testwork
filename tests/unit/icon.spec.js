import Icon from '@/components/Icon/index.vue';
import { mount } from '@vue/test-utils';

describe('Icon component', () => {
  it('1. Rendered success', async () => {
    const wrapper = await mount(Icon);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('2. Render svg inside success', async () => {
    const wrapper = await mount(Icon, {
      propsData: {
        icon: 'delete-outline',
      },
    });

    const svg = wrapper.find('svg');

    expect(svg.exists()).toBe(true);
  });
});
