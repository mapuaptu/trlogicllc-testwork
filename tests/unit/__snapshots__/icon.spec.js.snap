// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component 1. Rendered success 1`] = `
<div>
  <!---->
</div>
`;
